packages/transformers/markdown/src/MarkedTransformer.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+import {Transformer} from '@parcel/plugin';
+import {promisify} from '@parcel/utils';
+import marked from 'marked';
+import hljs from 'highlight.js';
+import fm from 'front-matter';
+import Mustache from 'mustache';
+
+const DEFAULT_TEMPLATE = `<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>{{title}}</title>
+</head>
+<body>
+  {{{ body }}}
+</body>
+</html>`;
+
+const markedParse = promisify(marked.parse);
+
+export default new Transformer({
+  async transform({asset, resolve, options}) {
+    asset.type = 'html';
+
+    let code = await asset.getCode();
+    let {body, attributes} = fm(code);
+
+    attributes.body = await markedParse(body, {
+      renderer: new marked.Renderer(),
+      highlight: (code: string, lang: string) => {
+        return hljs.highlight(lang, code).value;
+      },
+      pedantic: false,
+      gfm: true,
+      breaks: false,
+      sanitize: false,
+      smartLists: true,
+      smartypants: false,
+      xhtml: false,
+    });
+
+    let template = DEFAULT_TEMPLATE;
+    if (attributes.template) {
+      let templateLocation = await resolve(asset.filePath, attributes.template);
+      template = await options.inputFS.readFile(templateLocation, 'utf-8');
+      asset.addIncludedFile({
+        filePath: templateLocation,
+      });
+    }
+
+    let res = Mustache.render(template, attributes);
```

Comment:
This should probably also change imports to be relative to the markdown file location as otherwise imports will break

Reply:
The only (worse?) alternative would be to process the two assets separately and only then merge them in a packager...

Reply:
That would perhaps be faster but not entirely sure how that would work as the packager is the htmlpackager and not sure if we should add markdown stuff to it? I haven't written any packager stuff for parcel 2 yet so I might be wrong here

Reply:
I would do it like this:
The markdown transformer adds the html template as a dependency and then keeps the asset type at `md`.
Then a md packager takes the html dependency of the markdown asset (from the BundleGraph) and merges them.
This way, the template is also only transformed once.

(Maybe the bundler plugin needs to know about the md-html relationship as well.)
(The HTML transformer needs to be configure to not destroy the template tags.)

Reply:
sounds doable I'll look into it tomorrow